zoo/repos/tasks.py

```diff
@@ -82,3 +87,75 @@ def pull(reference, provider):
 
         repo_analyzers.run_all(repository, repo_path)
         runner.run_checks_and_save_results(AUDITING_CHECKS, repository, repo_path)
+
+
+@shared_task
+def sync_zoo_file():
+    for project in itertools.chain(
+        get_github_repositories(), get_gitlab_repositories()
+    ):
+        if settings.SYNC_REPOS_SKIP_FORKS and project["is_fork"]:
+            continue
+        update_project_from_zoo_file.apply_async(args=project)
+
+
+@shared_task
+def update_project_from_zoo_file(proj: Dict) -> None:
+    try:
+        content = get_zoo_file_content(proj)
+    except FileNotFoundError as err:
+        log.info("repos.sync_zoo_yml.file_not_found", error=err)
+    else:
+        if not validate(content):
+            return
+        update_or_create_service(parse(content), proj)
+
+
+def update_or_create_service(data: Dict, proj: Dict) -> None:
+    # Skip processing the zoo file if the repository is not yet synced
+    try:
+        repository = Repository.objects.get(
+            remote_id=int(proj["id"]), provider=proj["provider"]
+        )
+    except Repository.DoesNotExist:
+        return
+
+    service_defaults = {
+        "impact": data["impact"],
+        "status": data["status"],
+        "repository": repository,
+        "docs_url": data["docs_url"],
+        "slack_channel": data["slack_channel"],
+        "sentry_project": data["sentry_project"],
+        "sonarqube_project": data["sonarqube_project"],
+        "pagerduty_url": data["pagerduty_url"],
+        "tags": data["tags"],
+    }
+
+    service, _ = Service.objects.update_or_create(
+        owner=data["owner"], name=data["name"], defaults=service_defaults
+    )
+
+    # Delete all environments as yaml file has precedence
+    Environment.objects.filter(service=service).delete()
+
+    # Add all environments
+    for env in data["environments"]:
+        e = Environment(service=service, name=env["name"])
+        e.dashboard_url = env["dashboard_url"]
+        e.service_urls = env["service_urls"]
+        e.health_check_url = env["health_check_url"]
+        e.save()
+
+
+def get_zoo_file_content(proj: Dict) -> Union[str, None]:
+    if proj["provider"] == "gitlab":
```

Comment:
You should probably use `zoo.repos.utils.get_scm_module` here:

```python
provider = get_scm_module(proj["provider"])
provider.get_file_content(...)
```

Reply:
Good catch. I'll change it.